3rd-party/woocommerce-services.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class WC_Services_Installer {
+
+	public function init() {
+		$this->add_error_notice();
+		$this->try_install();
+	}
+
+	public function try_install() {
+		$action = filter_input( INPUT_GET, 'wc-services-action' );
```

Comment:
I believe [`filter_input()`](https://secure.php.net/manual/en/function.filter-input.php) was chosen because it handles the array key not existing - removing the need for `isset( $_GET[ 'wc-services-action' ] )`.

For non-string types, it can provide a mechanism for filtering the value upon retrieval as well.